qr-code/src/test/java/com/adyen/checkout/qrcode/DefaultQRCodeDelegateTest.kt

```diff
@@ -265,6 +293,17 @@ internal class DefaultQRCodeDelegateTest(
         }
     }
 
+    @Test
+    fun `download qr image successfully`(): Unit = runBlocking {
+        val expectedResult = Result.success(Unit)
+        whenever(fileDownloader.download(anyString(), anyString(), anyString(), anyString())) doReturn expectedResult
+
+        val actualResult = delegate.downloadQRImage()
+
+        assertEquals(expectedResult.isSuccess, actualResult.isSuccess)
```

Comment:
This line checks if `expectedResult.isSuccess == actualResult.isSuccess`, it seems like you only want to be sure that the success state is the same.
So for example if they are both `false` your test passes even though the test case is about download a QR image successfully.
What you want to check is that the whole result propagated as it is from `fileDownloader` to the `delegate` right?

Reply:
nice catch though